invenio_records_resources/services/records/queryparser/transformer.py

```diff
@@ -24,24 +24,30 @@
 class SearchFieldTransformer(TreeTransformer):
     """Transform from user-friendly field names to internal field names."""
 
-    def __init__(self, mapping, *args, **kwargs):
+    def __init__(self, mapping, safelist, *args, **kwargs):
```

Comment:
This can also be implemented in the `QueryParser` as a "step" of the parsing process. 
I added it here for now since we're visiting one field at the time (using `luqum`). So it was easier to do it here and didn't feel completely wrong.

But please challenge it :)